airflow/sensors/base.py

```diff
@@ -257,11 +258,14 @@ def _get_next_poke_interval(
 
     def prepare_for_execution(self) -> BaseOperator:
         task = super().prepare_for_execution()
+
         # Sensors in `poke` mode can block execution of DAGs when running
         # with single process executor, thus we change the mode to`reschedule`
         # to allow parallel task being scheduled and executed
-        if conf.get("core", "executor") == "DebugExecutor":
-            self.log.warning("DebugExecutor changes sensor mode to 'reschedule'.")
+        executor_name = conf.get("core", "executor")
+        executor = ExecutorLoader.load_executor(executor_name)
```

Comment:
You can just use `import_default_executor_cls` here I reckon

Reply:
@o-nikolas Thanks updated.